packages/runtime/src/types.ts

```diff
@@ -0,0 +1,20 @@
+import * as D from 'docmaps-sdk'
+import * as TE from 'fp-ts/TaskEither'
+
+// TODO: use io-ts? this is only decodable by consumers...
+export type ApiInfo = {
+  api_url: string
+  api_version: string
+  ephemeral_document_expiry: {
+    max_seconds: number
+    max_retrievals: number
+  }
+  peers: {
+    api_url: string
+  }[]
+}
+
+// TODO is this the same as the Client?
+export interface BackendAdapter {
+  docmapWithIri(iri: string): TE.TaskEither<Error, D.DocmapT>
+}
```

Comment:
If it's concievable that users would eventually write their own unsupported adapters it may be worthwhile to have this return a vanilla promise instead of an fp-ts datatype.